src/charm.py

```diff
@@ -95,17 +113,62 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
             self.unit.status = WaitingStatus("Waiting for vault initialization secret")
             event.defer()
             return
+        if not self.unit.is_leader() and len(self._other_peer_unit_addresses()) == 0:
```

Comment:
The spec states that the Vault leader may not be the same as the Juju leader. Should this be considered at this point?

Same question for the _on_install method.

Reply:
By "may" not I mean that we don't control this, it could be it or not but that is decided through raft's concensus algorithm, not the charm.